core/services/registrysyncer/syncer.go

```diff
@@ -35,9 +35,12 @@ type Syncer interface {
 }
 
 type registrySyncer struct {
-	stopCh    services.StopChan
-	launchers []Launcher
-	reader    types.ContractReader
+	stopCh          services.StopChan
+	launchers       []Launcher
+	reader          types.ContractReader
+	initReader      func(ctx context.Context, lggr logger.Logger, relayer contractReaderFactory, registryAddress string) (types.ContractReader, error)
```

Comment:
/nit: it doesn't seem like this can be set by any caller. iiuc, then there would be less indirection to simply make it a static method on the syncer, or omit it entirely and call newReader in it's place

Reply:
I added this mainly to make testing easier, such that a test could set initReader and set up a mock rather than setting up an actual chain reader.